### ISSUE TITLE
bug: live-reload not working when using adb reverse

### ISSUE THREAD

**Issue by czabaj** [ionic-team/capacitor#4125]
## Bug Report

App set for live-reload with

```
	"android": {
		"webContentsDebuggingEnabled": true
	},
	"server": {
		"cleartext": true,
		"url": "http://localhost:8080"
	}
```

The app runs in the simulator, displays the "Using app server http://localhost:8080" toast, but it serves static assets from the build folder and the live-reload really does not work.

I start the Android studio with 

```
yarn cap sync android && yarn cap copy android && yarn cap open android
```

### Capacitor Version

```
💊   Capacitor Doctor  💊

Latest Dependencies:

  @capacitor/cli: 2.4.6
  @capacitor/core: 2.4.6
  @capacitor/android: 2.4.6
  @capacitor/electron: 2.4.6
  @capacitor/ios: 2.4.6

Installed Dependencies:

  @capacitor/cli 2.4.5
  @capacitor/core 2.4.5
  @capacitor/ios 2.4.5
  @capacitor/android 2.4.5
  @capacitor/electron not installed

[success] Android looking great! 👌
  Found 9 Capacitor plugins for ios:
    capacitor-dark-mode (1.0.5)
    capacitor-plugin-ios-swipe-back (1.0.3)
    capacitor-rate-app (0.1.1)
    capacitor-secure-storage-plugin (0.5.0)
    cordova-plugin-appcenter-crashes (0.5.1)
    cordova-plugin-appcenter-shared (0.5.1)
    cordova-plugin-fingerprint-aio (4.0.2)
    cordova-plugin-inappbrowser (5.0.0-dev)
    onesignal-cordova-plugin (2.11.2)
[success] iOS looking great! 👌
```

### Platform(s)

Android

### Other Technical Details

`yarn --version` output: `1.22.10`

`node --version` output: `v12.19.0`

**Comment by jcesarmobile** [comment]
see https://github.com/ionic-team/capacitor/discussions/4123#discussioncomment-315565

**Comment by czabaj** [comment]
@jcesarmobile oh, I see. Thank you!

Previously, setting `localhost` worked because we set

```
adb reverse tcp:8080 tcp:8080
```

for the simulator or device.

But recently, this stopped working and the fact, that the simulator uses static files from `public/` dir instead, cost me a lot of time since it pretends it is working even showing the "Serving from ..." toast.

I set the server to `0.0.0.0` instead of the `localhost` and now it works again (after the `adb reverse ...` command).

I like the approach with `adb reverse` because we can share the `server` configuration in my team regardless of our network IP and, also, we run locally a mocking API, which we also need to reach on a different port and we can share the API config with web development as well with in-device development if we just `adb reverse` all the other ports.

I think it would be better when the build with `server` configuration will fail when the `server` is unreachable, serving the static files as a fallback is very confusing.

**Comment by czabaj** [comment]
Partially relates #4028

**Comment by jcesarmobile** [comment]
if you are using `adb reverse tcp:8080 tcp:8080` then it should work

**Comment by jcesarmobile** [comment]
closing as the PR was merged, the fix will be available on next release (2.4.7)

**Comment by Hamata6** [comment]
@jcesarmobile I'm having the same issue with adb reverse not live reloading. Any idea when 2.4.7 will be released? Thanks for fixing btw
